edb/edgeql/compiler/options.py

```diff
@@ -74,6 +75,10 @@ class GlobalCompilerOptions:
     #: to contain DML in the top-level shape computables.
     allow_top_level_shape_dml: bool = False
 
+    #: A list of DML expressions (statements and DML-containing
+    #: functions) that appear in a function body.
+    dml_exprs: List[qlast.Base] = dc_field(default_factory=list)
```

Comment:
Please move this to `Environment`.  `CompilerOptions` are options passed _to_ the compiler, they shouldn't be used to pass information back to the caller or mutated.

Reply:
Done